contracts/deploy/mainnet/119_pool_booster_curve.js

```diff
@@ -10,54 +13,118 @@ module.exports = deploymentWithGovernanceProposal(
     deployerIsProposer: false,
     proposalId: "",
   },
-  async ({ deployWithConfirmation, withConfirmation }) => {
+  async ({ withConfirmation }) => {
     const { deployerAddr } = await getNamedAccounts();
     const sDeployer = await ethers.provider.getSigner(deployerAddr);
-    console.log(`Using deployer account: ${deployerAddr}`);
-    // 1. Deploy proxy
-    const dCurvePoolBoosterProxy = await deployWithConfirmation(
-      "CurvePoolBoosterProxy"
+    console.log(`Deployer address: ${deployerAddr}`);
+
+    const cCurvePoolBooster = await ethers.getContractAt(
+      "CurvePoolBooster",
+      addresses.zero
     );
-    const cCurvePoolBoosterProxy = await ethers.getContract(
-      "CurvePoolBoosterProxy"
+    const cCurvePoolBoosterProxy = await ethers.getContractAt(
+      "CurvePoolBoosterProxy",
+      addresses.zero
     );
 
-    // 2. Deploy implementation
-    const dCurvePoolBoosterImpl = await deployWithConfirmation(
-      "CurvePoolBooster",
-      [
-        42161, // Arbitrum chain id
-        addresses.mainnet.CampaignRemoteManager, // Campaign Remote Manager (VotemarketV2 entry point)
-        addresses.mainnet.OUSDProxy, // Bribe token
-        addresses.mainnet.CurveOUSDUSDTGauge, // Gauge
-      ]
+    console.log("\nStarting deployment using CreateX");
+    const rewardToken = addresses.mainnet.OUSDProxy;
+    const gauge = addresses.mainnet.CurveOUSDUSDTGauge;
+    const targetedChainId = 42161; // arbitrum
+    const fee = 0;
+
+    // Get CreateX contract
+    const cCreateX = await ethers.getContractAt(createxAbi, addresses.createX);
+
+    // Generate encoded salt (deployer address || crossChainProtectionFlag || bytes11(keccak256(rewardToken, gauge)))
+    const addressDeployerBytes20 = ethers.utils.hexlify(
```

Comment:
Can we separate the salt creation code into a deploy utilities source file? It is going to be used in many deploy files and it should improve readability and risk of making mistakes.

Reply:
Done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2360/commits/f2bf316b218ef98eb93cfbe10bf13a599169ffdf